napari/_vispy/canvas.py

```diff
@@ -71,26 +163,293 @@ def _set_theme_change(self, theme: str):
 
     @bgcolor.setter
     def bgcolor(self, value):
         _value = self._background_color_override or value
-        SceneCanvas.bgcolor.fset(self, _value)
+        self._scene_canvas.bgcolor = _value
 
     @property
     def central_widget(self):
         """Overrides SceneCanvas.central_widget to make border_width=0"""
-        if self._central_widget is None:
-            self._central_widget = Widget(
-                size=self.size, parent=self.scene, border_width=0
+        if self._scene_canvas._central_widget is None:
+            self._scene_canvas._central_widget = Widget(
+                size=self.size,
+                parent=self._scene_canvas.scene,
+                border_width=0,
             )
-        return self._central_widget
+        return self._scene_canvas._central_widget
 
-    def _process_mouse_event(self, event):
-        """Ignore mouse wheel events which have modifiers."""
-        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+    @property
+    def size(self):
+        """Return canvas size as tuple (height, width) or accepts size as tuple (height, width)
+        and sets Vispy SceneCanvas size as (width, height)."""
+        return self._scene_canvas.size[::-1]
+
+    @size.setter
+    def size(self, size):
+        self._scene_canvas.size = size[::-1]
+
+    @property
+    def cursor(self) -> QCursor:
+        """Cursor associated with native widget"""
+        return self.native.cursor()
+
+    @cursor.setter
+    def cursor(self, q_cursor):
+        """Setting the cursor of the native widget"""
+        self.native.setCursor(q_cursor)
+
+    def delete(self):
+        """Schedules the native widget for deletion"""
+        self.native.deleteLater()
+
+    def _on_interactive(self):
+        """Link interactive attributes of view and viewer."""
+        self.view.interactive = self.viewer.camera.interactive
+
+    def _map_canvas2world(self, position):
+        """Map position from canvas pixels into world coordinates.
+
+        Parameters
+        ----------
+        position : 2-tuple
+            Position in canvas (x, y).
+
+        Returns
+        -------
+        coords : tuple
+            Position in world coordinates, matches the total dimensionality
+            of the viewer.
+        """
+        nd = self.viewer.dims.ndisplay
+        transform = self.view.scene.transform
+        mapped_position = transform.imap(list(position))[:nd]
+        position_world_slice = mapped_position[::-1]
+
+        # handle position for 3D views of 2D data
+        nd_point = len(self.viewer.dims.point)
+        if nd_point < nd:
+            position_world_slice = position_world_slice[-nd_point:]
+
+        position_world = list(self.viewer.dims.point)
+        for i, d in enumerate(self.viewer.dims.displayed):
+            position_world[d] = position_world_slice[i]
+
+        return tuple(position_world)
+
+    def _process_mouse_event(self, mouse_callbacks, event):
```

Comment:
I see really nice type annotation in docstring. Why are they not added to the function signature?